python/task_manager/worker_thread.py

```diff
@@ -57,6 +55,7 @@ def shut_down(self):
         """
         Shut down the thread and wait for it to exit before returning
         """
+        self._results_poller = None
```

Comment:
I may have inherited Alan's paranoia about cleaning up references to Qt objects. I'll remove it and run my tests to see what happens.